src/view/selection.js

```diff
@@ -16,21 +16,23 @@ import Node from './node';
 import Element from './element';
 import count from '@ckeditor/ckeditor5-utils/src/count';
 import isIterable from '@ckeditor/ckeditor5-utils/src/isiterable';
+import DocumentSelection from './documentselection';
```

Comment:
I remember, that in the model I had some problem with it. The two-way importing with extending other class AFAIR. (In the model there is also a `LiveSelection` which inherits from the `Selection`).

Reply:
I've added there something like duck typing to prevent this. But If there's no problem with the current implementation, it's not worth to change it.